### ISSUE TITLE
Confusing unused_attributes warning for no_mangle on extern "C" static.

### ISSUE THREAD

**Issue by glandium** [rust-lang/rust#79619]
In Firefox, there was code like this:
```
extern "C" {
  #[no_mangle]
  static gWinEventLogSourceName: *const u16;
}
```
This triggered the following warning:
```
error: attribute should be applied to a function or static
   --> toolkit/mozapps/defaultagent/rust/src/lib.rs:111:5
    | 
111 |     #[no_mangle]
    |     ^^^^^^^^^^^^
112 |     static gWinEventLogSourceName: *const u16; 
    |     ------------------------------------------ not a function or static
    | 
    = note: `-D unused-attributes` implied by `-D warnings`
    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
error: aborting due to previous error
```

The confusing part is that it says "not a function or static" for something that is a static, albeit extern.

The no_mangle _is_ unnecessary in this case, but coming to that conclusion from the warning message is not trivial.

**Comment by jonas-schievink** [comment]
Duplicate of https://github.com/rust-lang/rust/issues/78989
